app/novelty/compute_novelty.py

```diff
@@ -377,98 +463,412 @@ def novelty_score(fda_status, recency, similarity):
+                                    if len(phase_trials) > 0:
+                                        phase_trials.sort(reverse=True)
+                                        df_numpy[idi].extend([phase_trials[0]])
+                                        if "4" in phase_trials[0]:
+                                            novelty_score_rec_clin.append(0)
+                                            novelty_score_md[idi] = 0
+                                        elif "3" in phase_trials[0]:
+                                            novelty_score_rec_clin.append(novelty_score_rec[idi]*0.2)
+                                        elif "2" in phase_trials[0]:
+                                            novelty_score_rec_clin.append(novelty_score_rec[idi]*0.6)
+                                        elif "1" in phase_trials[0]:
+                                            novelty_score_rec_clin.append(novelty_score_rec[idi]*0.8)
+                                        else:
+                                            novelty_score_rec_clin.append(novelty_score_rec[idi])
+                                        found = 1
+                                        # print(idi, phase_trials)
+                            if found == 0:
+                                df_numpy[idi].extend([f"No clinical information"])
+                                novelty_score_rec_clin.append(novelty_score_rec[idi])
+
+                        else:
+                            df_numpy[idi].extend([f"No clinical information"])
+                            novelty_score_rec_clin.append(novelty_score_rec[idi])
+
+                else:
+                    df_numpy[idi].extend([f"Incorrect Category of Result"])
+                    df_numpy[idi].extend([f"Incorrect Category of Result"])
+                    novelty_score_md.append(0)
+                    novelty_score_rec_clin.append(0)
+            for c in range(len(novelty_score_rec)):
+                novelty_score_final = wt_rec_clin * novelty_score_rec_clin[c] + wt_md * novelty_score_md[c]
+                df_numpy[c].append(novelty_score_final)
+
+        df_numpy = pd.DataFrame(df_numpy, columns= column_list)
+
+    except Exception as e:
+        print(f"Error Encountered: {e}")
+        if result_id_cat in ["biolink:SmallMolecule", "biolink:Drug", "biolink:ChemicalEntity", "biolink:MolecularMixture", "biolink:MolecularEntity"]:
+            column_list = ['Query ID', 'Result ID', 'Location 1', 'Location 2', 'Curated or not ?', 'Recency Score', 'Molecular Similarity', 'Clinical Information', 'novelty_score']
+        elif result_id_cat in ["biolink:Protein", "biolink:Gene"]:
+            column_list = ['Query ID', 'Result ID', 'Location 1', 'Location 2', 'Curated or not ?', 'Recency Score',
+                           'Gene Distinctiveness', 'TDLs', 'novelty_score']
+        df_numpy = pd.DataFrame([["NAN"]*len(column_list)]*len(message['results']), columns = column_list)
+
+    return df_numpy
```

Comment:
The final output used to only have `drug` and `novelty_score`, with `drug` being the curie. Is that now `Result ID`? Also, at least for the query I'm trying, every value is NaN. If that's expected, could we make the default value be 0 or something?

Reply:
Yes. Since now we have MVP1 and MVP2 queries I changed the 'drug' to 'Result ID'. Can you send me the query? The lowest value is 0 for the novelty score. If it's NaN then there might be an issue I'd like to look a bit closer into.

Reply:
I'm working with a smaller message, and a large message. The smaller message seems to be fine, but the larger message returns all NaNs. The CI PK for that larger message is d1e6d43f-cb3e-4c6b-a6bc-5393f576053f.

Reply:
Hi Max,

Reply:
The 422 error was given by the ARS, but that's not necessarily indicative of what the Appraiser did. I can pass that message to the Appraiser and it handles it fine, other than the novelty code, so I think something needs to be fixed in this code.

FYI, ALL fields are set to NaN, not just novelty_score.
FYI FYI, if the novelty code errors out, I'm already setting the default novelty to 0.0, so I think it would be a good idea to have that same default value in your code instead of a NaN (which I think is actually the string "NaN" instead of the value NaN.

Reply:
Ok. I've changed the default NANs (for instances when an error is encountered) to 0 for all entries in the DataFrame generated.

Reply:
So now the default value is 0, but ALL fields are still 0. I'm also not seeing any error messages come out of novelty, so it looks like everything went fine. In my opinion, `Query ID` and `Result ID` should always have values and not be 0, regardless of any errors.